documentation/proposals/Proposal - Region Specific Key Input.md

```diff
@@ -0,0 +1,141 @@
+# Summary
+- Make the windowing API return the key actually pressed, or any other key for that matter, instead of the key that is at
+the same location on a US-Keyboard.
+
+# Contributors
+- Matija Brown, None
+
+# Current Status
+- [x] Proposed
+- [ ] Discussed with API Review Board (ARB)
+- [ ] Approved
+- [ ] Implemented
+
+# Design Decissions
+- The main aspect is to change the original API as little as possible and also make the region-adapting
+toggleable / add the abillity to manually set a keyboard layout. To achieve this diversity,
+new layouts can be imported from an external file, but the included ones are hard coded. This way one can easilly switch between
+layouts and / or import a new, custome one while the rest of the input API remains unchanged. Due to the fact this is toggleable,
+it will also work the same way as before. The layout is saved in the IInputContext, as it is not possible to have different
+keyboard layouts on different keyboards at the same time, so it needn't be saved in every keyboard instance. If writing custom layouts
+turns out to be a common thing, there will be a layout generator created as well. For now key modifiers such as SHIFT will not be
+considered, but that might be implemented at a later time.
+
+# Proposed API
+
+## IKeyboardLayout
+```cs
+internal interface IKeyboardLayout
+{
+
+    /// <summary>
+    /// Gets the name from an instance.
+    /// </summary>
+    string Name { get; }
+
+    /// <summary>
+    /// Called from GlfwKeyboard.ConvertKey, easy place to get the keys,
+    /// and no use converting to Silk.NET keys and then swap them arround again.
+    /// </summary>
+    Key MapKey(Keys glfwKey);
+
+}
+```
+
+## QWERTYLayout
+```cs
+/// <summary>
+/// Example class for all layouts.
+/// </summary>
+public sealed class QWERTYLayout : IKeyboardLayout
+{
+
+    /// <summary>
+    /// The name of the layout.
+    /// </summary>
+    public const string NAME = "QWERTY";
+
+    /// <summary>
+    /// Non-Static name.
+    /// </summary>
+    public string Name { get } = NAME;
```

Comment:
`Name => NAME;` is likely preferable.
Not sure about the const, a property like that is quasi-constant anyways, but I guess for attributes?

Reply:
Well, yes. But I wanted to non-constant property and I believe it's good practice to not directly set that field in the inbuilt layouts.